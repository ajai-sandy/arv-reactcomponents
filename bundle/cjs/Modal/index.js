"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _reactDom = require("react-dom");

var _reactDom2 = _interopRequireDefault(_reactDom);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _ModalContainer = require("../ModalContainer");

var _ModalContainer2 = _interopRequireDefault(_ModalContainer);

var _methods = require("./methods");

var _methods2 = _interopRequireDefault(_methods);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function Modal(props) {
  var isOpen = props.isOpen,
      otherProps = _objectWithoutProperties(props, ["isOpen"]);

  if (isOpen) {
    return _reactDom2.default.createPortal(_react2.default.createElement(_ModalContainer2.default, otherProps), document.querySelector(".nwc-modal-holder"));
  }
  return null;
}

(0, _methods2.default)();

Modal.isModalOpen = _ModalContainer2.default.isModalOpen;
Modal.propTypes = {
  isOpen: _propTypes2.default.bool.isRequired
};

exports.default = Modal;