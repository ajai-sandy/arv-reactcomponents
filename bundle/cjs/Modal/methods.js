"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
function createBaseContainer() {
  var modalContainer = document.createElement("div");
  modalContainer.className = "nwc-modal-holder";
  if (document.querySelectorAll("body")[0]) {
    document.querySelectorAll("body")[0].appendChild(modalContainer);
  }
}

exports.default = createBaseContainer;