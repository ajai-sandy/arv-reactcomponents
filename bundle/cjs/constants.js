"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var screenMedia = exports.screenMedia = {
  lg: 1200,
  md: 992,
  sm: 768
};

var isMobile = exports.isMobile = window.innerWidth <= 768;
var localStorage = exports.localStorage = "localStorage" in window && window.localStorage || null;

var windowRequestAnimationFrame = exports.windowRequestAnimationFrame = window.requestAnimationFrame;
var windowCancelAnimationFrame = exports.windowCancelAnimationFrame = window.cancelAnimationFrame;

var vendorsPrefixes = exports.vendorsPrefixes = ["ms", "moz", "webkit", "o"];
var vendorRequestAnimationFrame = exports.vendorRequestAnimationFrame = vendorsPrefixes.find(function (prefix) {
  return window[prefix + "RequestAnimationFrame"];
});
var vendorCancelAnimationFrame = exports.vendorCancelAnimationFrame = vendorsPrefixes.find(function (prefix) {
  return window[prefix + "CancelAnimationFrame"] || window[prefix + "CancelRequestAnimationFrame"];
});