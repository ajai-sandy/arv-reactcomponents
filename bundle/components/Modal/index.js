import React from "react";
import ReactDOM from "react-dom";
import PropTypes from "prop-types";
import ModalContainer from "../ModalContainer";

import createBaseContainer from "./methods";

function Modal(props) {
  const { isOpen, ...otherProps } = props;

  if (isOpen) {
    return ReactDOM.createPortal(
      <ModalContainer {...otherProps} />,
      document.querySelector(".nwc-modal-holder"),
    );
  }
  return null;
}

createBaseContainer();

Modal.isModalOpen = ModalContainer.isModalOpen;
Modal.propTypes = {
  isOpen: PropTypes.bool.isRequired,
};

export default Modal;
