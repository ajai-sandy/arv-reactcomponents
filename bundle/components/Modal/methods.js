function createBaseContainer() {
  const modalContainer = document.createElement("div");
  modalContainer.className = "nwc-modal-holder";
  if (document.querySelectorAll("body")[0]) {
    document.querySelectorAll("body")[0].appendChild(modalContainer);
  }
}

export default createBaseContainer;
