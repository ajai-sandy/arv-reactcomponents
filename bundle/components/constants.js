export const screenMedia = {
  lg: 1200,
  md: 992,
  sm: 768,
};

export const isMobile = window.innerWidth <= 768;
export const localStorage =
  ("localStorage" in window && window.localStorage) || null;

export const windowRequestAnimationFrame = window.requestAnimationFrame;
export const windowCancelAnimationFrame = window.cancelAnimationFrame;

export const vendorsPrefixes = ["ms", "moz", "webkit", "o"];
export const vendorRequestAnimationFrame = vendorsPrefixes.find(
  prefix => window[`${prefix}RequestAnimationFrame`],
);
export const vendorCancelAnimationFrame = vendorsPrefixes.find(
  prefix =>
    window[`${prefix}CancelAnimationFrame`] ||
    window[`${prefix}CancelRequestAnimationFrame`],
);
