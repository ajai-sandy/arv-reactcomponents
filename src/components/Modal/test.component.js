import React, { Component } from "react";

// Import common components
import { Modal } from "..";

class ModalTest extends Component {
  constructor(props) {
    super(props);

    this.state = {
      showHaiIsOpen: false,
    };

    this.showHaiMsg = this.showHaiMsg.bind(this);
  }

  showHaiMsg(bool) {
    this.setState({
      showHaiIsOpen: bool,
    });
  }

  render() {
    const { showHaiIsOpen } = this.state;

    return (
      <Modal
        className="nw-modal"
        isOpen={showHaiIsOpen}
        onClose={() => {
          this.showHaiMsg(false);
        }}
      >
        <div>Hai</div>
      </Modal>
    );
  }
}

export default ModalTest;
