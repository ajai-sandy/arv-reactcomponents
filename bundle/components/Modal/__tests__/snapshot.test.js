import React from "react";
import { mount, shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";

import TestComponent from "../test.component";
import * as constants from "../../constants";

import Modal from "../";

describe("Root Component Tests", () => {
  beforeAll(() => {
    Object.defineProperty(global.window.history, "state", {
      value: {},
    });
  });

  it("Matches Shallow SnapShot", () => {
    expect(shallowToJson(shallow(<TestComponent />))).toMatchSnapshot();
  });

  it("Matches Mounted SnapShot", done => {
    const mounted = mount(<TestComponent />);

    expect(Modal.isModalOpen()).toBe(false);
    mounted.instance().showHaiMsg(true);
    expect(Modal.isModalOpen()).toBe(true);

    setTimeout(() => {
      mounted.update();
      expect(mounted).toMatchSnapshot();
      mounted.find("button.nwc-close").simulate("click");

      setTimeout(() => {
        done();
      }, 300);
    }, 300);
  });

  it("Matches Mounted SnapShot Mobile", done => {
    constants.isMobile = true;
    const mounted = mount(<TestComponent />);

    expect(Modal.isModalOpen()).toBe(false);
    mounted.instance().showHaiMsg(true);
    expect(Modal.isModalOpen()).toBe(true);

    setTimeout(() => {
      mounted.update();
      expect(mounted).toMatchSnapshot();
      mounted.find("button.nwc-close").simulate("click");

      setTimeout(() => {
        done();
      }, 300);
    }, 300);
  });

  afterEach(() => {
    constants.isMobile = false;
  });
});
