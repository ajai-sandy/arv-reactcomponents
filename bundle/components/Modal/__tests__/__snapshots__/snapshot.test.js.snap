// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root Component Tests Matches Mounted SnapShot 1`] = `
<ModalTest>
  <Modal
    className="nw-modal"
    isOpen={true}
    onClose={[Function]}
  >
    <ModalContainer
      className="nw-modal"
      onClose={[Function]}
      preventfix={false}
    >
      <div
        className="nwc-modal-container nw-modal"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="nwc-modal"
          onClick={[Function]}
          onKeyDown={null}
          role="presentation"
        >
          <Close
            className="nwc-close-normal nwc-modal-close"
            onClick={[Function]}
          >
            <Button
              className="nwc-close nwc-close-normal nwc-modal-close"
              onClick={[Function]}
            >
              <button
                className="nwc-btn nwc-close nwc-close-normal nwc-modal-close"
                onClick={[Function]}
                tabIndex="0"
              >
                ×
              </button>
            </Button>
          </Close>
          <div
            className="nwc-custom-scrollbar nwc-modal-children"
          >
            <div>
              Hai
            </div>
          </div>
        </div>
      </div>
    </ModalContainer>
  </Modal>
</ModalTest>
`;

exports[`Root Component Tests Matches Mounted SnapShot Mobile 1`] = `
<ModalTest>
  <Modal
    className="nw-modal"
    isOpen={true}
    onClose={[Function]}
  >
    <ModalContainer
      className="nw-modal"
      onClose={[Function]}
      preventfix={false}
    >
      <div
        className="nwc-modal-container nw-modal"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="nwc-modal"
          onClick={[Function]}
          onKeyDown={null}
          role="presentation"
        >
          <Close
            className="nwc-close-normal nwc-modal-close"
            onClick={[Function]}
          >
            <Button
              className="nwc-close nwc-close-normal nwc-modal-close"
              onClick={[Function]}
            >
              <button
                className="nwc-btn nwc-close nwc-close-normal nwc-modal-close"
                onClick={[Function]}
                tabIndex="0"
              >
                ×
              </button>
            </Button>
          </Close>
          <div
            className="nwc-custom-scrollbar nwc-modal-children"
          >
            <div>
              Hai
            </div>
          </div>
        </div>
      </div>
    </ModalContainer>
  </Modal>
</ModalTest>
`;

exports[`Root Component Tests Matches Shallow SnapShot 1`] = `
<Modal
  className="nw-modal"
  isOpen={false}
  onClose={[Function]}
>
  <div>
    Hai
  </div>
</Modal>
`;
